job/job.go

```diff
@@ -63,23 +65,66 @@ type Job interface {
 	// return quickly.
 	Status() string
 
-	// Name returns the name of the job.
-	Name() string // @todo: change this to Id()
+	// Id returns the Id structure the job was created with. This is used primarily
+	// for status and logging by Spin Cycle. The Id values for the job must exactly
+	// match the Id values passed to the external job factory which created the job,
+	// else Spin Cycle returns ErrWrongJobId.
+	Id() Id
+}
+
+// Id represents how jobs are uniquely identified per request. Type and Name are
+// user-defined in the external job factory (EJF) and request spec, respectively.
+// Id is defined per request by Spin Cycle. An example for each value:
+//
+//   Type:  net/down-ip
+//   Name:  down-sip-on-eth0
+//   Id:    9m4e2mr0
+//
+// Job types are defined in code and in the external job factory (EJF). They
+// represent what the job is and does. Job names are defined in request specs;
+// they are named instantiantions of job types. The name should be short but
+// meaningful the the context of the request. Job IDs are short, random strings
+// generated by Spin Cycle. Id is only guaranteed to be unique per request, not
+// globally.
+type Id struct {
+	// User-defined low-level job type implemented in code and made by external
+	// job factory (EJF). Any value that the EJF is coded to make is valid. Job
+	// types are heavily reused by being given unique names in request specs.
+	Type string
 
-	// Type returns the type of the job.
-	Type() string
+	// User-defined job name in a request spec. Names must be unique per request.
```

Comment:
Probably more accurate to say that names must be unique per sequences, no? Two different sequences used within a request should both be able to have a job with the same name